pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1151 @@
+            "sampling_rate": self.acquisition_sampling_rate,
+            "grid_number": self._grid_number,
+            "status": self.acquisition_status,
+            "memory_size": self.memory_size,
+            "xdiv": self.timebase_scale,
+            "xoffset": self.timebase_offset
+        }
+        preamble["average"] = self.acquisition_average if preamble["type"][0] == "average" else None
+        strict_discrete_set(self.waveform_source, ["C1", "C2", "C3", "C4", "MATH"])
+        preamble["sampled_points"] = self.acquisition_sample_size(self.waveform_source)
+        if self.waveform_source == "MATH":
+            preamble["ydiv"] = self.math_vdiv
+            preamble["yoffset"] = self.math_vpos
+        else:
+            preamble["ydiv"] = self.ch(self.waveform_source).scale
+            preamble["yoffset"] = self.ch(self.waveform_source).offset
+        return preamble
+
+    def _digitize(self, source, requested_bytes=None):
+        """ Acquire waveforms according to the settings of the acquire commands.
+        Note.
+        If the requested number of bytes is not specified, the default chunk size is used,
+        but in such a case it cannot be quaranteed that the message is received in its entirety.
+        :param source: source of data: "C1", "C2", "C3", "C4", "MATH".
+        :param: requested_bytes: number of bytes expected from the scope (including the header
+        and footer).
+        :return: bytearray with raw data. """
+        if requested_bytes is None:
+            values = self.binary_values(f"{source}:WF? DAT2", dtype=np.uint8)
+        else:
+            with _ChunkResizer(self.adapter, requested_bytes):
+                values = self.binary_values(f"{source}:WF? DAT2", dtype=np.uint8)
+            read_bytes = len(values)
+            if read_bytes != requested_bytes:
+                raise ValueError(f"read bytes ({len(values)}) != "
+                                 f"requested bytes ({requested_bytes})")
+        return values
+
+    #################
+    # Download data #
+    #################
+
+    def download_image(self):
+        """ Get a BMP image of oscilloscope screen in bytearray of specified file format.
+        """
+        # Using binary_values query because default interface does not support binary transfer
+        with _ChunkResizer(self.adapter, 20 * 1024 * 1024):
+            img = self.binary_values("SCDP", dtype=np.uint8)
+        return bytearray(img)
+
+    def download_data(self, source, requested_points=0, sparsing=1, averaging=1):
+        """ Get data points from the specified source of the oscilloscope. Ther returned objects are
+        two np.ndarray of data and time points and a dict with the waveform preamble, that contains
+        metadata about the waveform.
+        Note.
+        :param source: measurement source. It can be "C1", "C2", "C3", "C4", "MATH.
+        :param requested_points: number of points to acquire. If 0 or None all available points
```

Comment:
```suggestion
        :param requested_points: number of points to acquire. If 0 or None, all available points
```

Reply:
Fixed in commit 35c2c674ffaa6814c9b474667611243933f53238